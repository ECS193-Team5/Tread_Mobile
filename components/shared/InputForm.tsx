import React, {useEffect, useState} from 'react';
import {TextInput, View} from "react-native";
import {InputFormStyle} from "../../css/shared/InputFormStyle";

function InputForm(props): JSX.Element {
  let initValue = "";
  let curStyle = InputFormStyle.InitInput;
  let textColor = '#9B9595';
  // const [curStyle, setCurStyle] = useState(InputFormStyle.InitInput)
  // const [textColor, setTextColor] = useState('#9B9595');

  useEffect(() => {
    initValue = props.value;
  },[])
  const isValid = (text) => {
    return text.length > 0
  }
  const handleTextChange = (text) => {
    if(initValue === text) {
      curStyle = InputFormStyle.InitInput;
      textColor = '#9B9595';
    } else {
      if(props.valid) {
        curStyle = InputFormStyle.ValidInput;
        textColor = '#014421';
      } else {
        curStyle = InputFormStyle.InvalidInput;
        textColor = '#C65656';
      }
    }

    props.setValid(isValid(text));
    props.setValue(text);
  }

  return (
    <View>
      <TextInput
        placeholder = {props.placeholder}
        style = {curStyle}
        placeholderTextColor = {textColor}
        editable = {props.editable}
        onChangeText={handleTextChange}
      />
    </View>
  )
}

export default InputForm;