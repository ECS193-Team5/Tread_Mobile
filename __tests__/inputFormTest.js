/**
 * @format
 */
import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer, {act} from 'react-test-renderer';
import InputForm from "../components/test/InputForm";
import { fireEvent, render, screen } from "@testing-library/react-native"
import { toHaveStyle } from "@testing-library/jest-native";
expect.extend({ toHaveStyle });
it('Correct Render', () => {
    let tree = null;

    act(() => {
        tree = renderer.create(
            <InputForm
                // value={"Yes"}
            />
        ).toJSON();
    })

    expect(tree).toMatchSnapshot();
});

it('Placeholder Set', () => {

    render(
        <InputForm
            placeholder={"Test Placeholder"}
        />
    );

    const element = screen.getByTestId('input')
    expect(element.props.placeholder).toBe("Test Placeholder");
});

it('Test init style', () => {

    render(
        <InputForm
            placeholder={"Test Placeholder"}
            value={""}
            multiline={false}
        />
    );

    const element = screen.getByTestId('input')

    expect(element).toHaveStyle({
        borderRadius: 32,
        borderColor: '#9B9595',
        borderWidth: 2,
        paddingLeft: 25,
        color: '#9B9595',
        height : '100%',
        width: '100%',
        fontSize : 16,
    })
});

