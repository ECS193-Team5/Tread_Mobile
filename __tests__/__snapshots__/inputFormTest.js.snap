// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correct Render 1`] = `null`;
